### PR TITLE
Onboarding branch needs to delete the step of updating target ci

### PR DESCRIPTION
**Pull request by sima-zhu**
Fixed the release pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=838480&view=results

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=838731&view=results

### CODE REVIEW

File: eng/common/pipelines/templates/steps/docs-metadata-release.yml [hunk 1]
```diff
@@ -75,22 +75,6 @@ steps:
...
-        -WorkDirectory "${{ parameters.WorkingDirectory }}"
-        -RepoId ${{ parameters.RepoId }}
-        -Repository ${{ parameters.PackageRepository }}
-        -ReleaseSHA ${{ parameters.ReleaseSha }}
-        -DocRepoLocation "${{ parameters.WorkingDirectory }}/repo"
-        -Configs "${{ parameters.CIConfigs }}"
-      pwsh: true
-    env:
-      GH_TOKEN: $(azuresdk-github-pat)
   - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/docs-metadata-release.yml:78]: Are the other steps needed after removing this? I suspect the others will always be a noop now.

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/docs-metadata-release.yml:78]: @danieljurek Is the onboarding branch still needed once we want to update the package list from release csv?

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/docs-metadata-release.yml:78]: Testing in .NET template

Comment (sima-zhu) [reply eng/common/pipelines/templates/steps/docs-metadata-release.yml]: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=838947&view=logs&j=5ccebfdc-da53-51d4-fecb-9480b00e8628&t=5ccebfdc-da53-51d4-fecb-9480b00e8628

**Review by weshaggard — APPROVED** [review]
Assuming you have finished testing the change looks good.

**Review by danieljurek — APPROVED** [review]

**Pull request merged**